### PR TITLE
Introduce code_hosts table

### PR DESCRIPTION
**Pull request by eseliger**
This is the first PR towards getting code hosts introduced as a concept to Sourcegraph.

Closes https://github.com/sourcegraph/sourcegraph/issues/55273

## Test plan

Verified on a PoC branch that this schema makes sense.

### CODE REVIEW

**Review by varsanojidan — APPROVED** [review]

File: migrations/frontend/1690460411_add_code_hosts_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE IF NOT EXISTS code_hosts (
```
Comment (varsanojidan) [inline migrations/frontend/1690460411_add_code_hosts_table/up.sql:1]: Do we also need to create the deleted_at virtual column here?

**Review by BolajiOlajide — APPROVED** [review]

**Review by sashaostrikov — APPROVED** [review]
LGTM! +1 to updated_at and deleted_at columns.

File: migrations/frontend/1690460411_add_code_hosts_table/up.sql (hunk 1 above)
Comment (eseliger) [reply migrations/frontend/1690460411_add_code_hosts_table/up.sql:1]: Nope, deleted_at is going to be a SQL condition on ListCodeHosts through:

```
	if !opts.IncludeDeleted {
		// Don't show code hosts for which all external services are soft-deleted or hard-deleted.
		conds = append(conds, sqlf.Sprintf("(EXISTS (SELECT 1 FROM external_services WHERE external_services.code_host_id = code_hosts.id AND deleted_at IS NULL))"))
	}
```

File: migrations/frontend/1690460411_add_code_hosts_table/up.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    id SERIAL PRIMARY KEY,
+    kind text NOT NULL,
+    url text NOT NULL UNIQUE,
+    api_rate_limit_quota integer,
+    api_rate_limit_interval_seconds integer,
+    git_rate_limit_quota integer,
+    git_rate_limit_interval_seconds integer,
+    created_at timestamp with time zone NOT NULL DEFAULT now()
```
[the remark this answers is not in the document]
Comment (eseliger) [reply migrations/frontend/1690460411_add_code_hosts_table/up.sql:9]: Sure, why not 👍

File: migrations/frontend/1690460411_add_code_hosts_table/up.sql (hunk 1 above)
Comment (eseliger) [reply migrations/frontend/1690460411_add_code_hosts_table/up.sql:1]: reason being, we don't want to allow to delete code hosts for now, we just don't want to show ones without external services that would be misleading.
Once we spend more time on the code host model, we will want to allow the deletion of an entire host, which will then in turn need to delete all of the external services, but not modeling that out yet :)

**Pull request merged**